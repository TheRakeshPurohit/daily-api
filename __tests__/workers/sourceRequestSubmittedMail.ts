import nock from 'nock';
import { expectSuccessfulBackground } from '../helpers';
import { sendEmail, User as GatewayUser } from '../../src/common';
import worker from '../../src/workers/sourceRequestSubmittedMail';

jest.mock('../../src/common/mailing', () => ({
  ...(jest.requireActual('../../src/common/mailing') as Record<
    string,
    unknown
  >),
  sendEmail: jest.fn(),
}));

beforeEach(async () => {
  jest.resetAllMocks();
});

const mockUsersMe = (user: GatewayUser): nock.Scope =>
  nock(process.env.GATEWAY_URL)
    .get('/v1/users/me')
    .matchHeader('authorization', `Service ${process.env.GATEWAY_SECRET}`)
    .matchHeader('user-id', user.id)
    .matchHeader('logged-in', 'true')
    .reply(200, user);

it('should send mail when the source request is submitted', async () => {
  const mockedUsers: GatewayUser[] = [
    {
      id: '1',
      email: 'lee@acme.com',
      name: 'Lee',
      image: 'https://daily.dev/lee.jpg',
      reputation: 5,
      permalink: 'https://daily.dev/lee',
    },
  ];
  mockedUsers.forEach(mockUsersMe);
  await expectSuccessfulBackground(worker, {
    userId: '1',
    createdAt: 1601187916999999,
    sourceName: 'Demo',
    sourceUrl: 'https://demo.com/sitemap.xml',
  });
  expect(sendEmail).toBeCalledTimes(1);
  expect(jest.mocked(sendEmail).mock.calls[0]).toMatchSnapshot();
});
