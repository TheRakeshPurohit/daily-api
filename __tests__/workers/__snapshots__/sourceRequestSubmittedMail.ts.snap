// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should send mail when the source request is submitted 1`] = `
Array [
  Object {
    "asm": Object {
      "groupId": 12850,
    },
    "category": "Notification",
    "dynamicTemplateData": Object {
      "first_name": "Lee",
      "rss_link": "https://demo.com/sitemap.xml",
      "source_name": "Demo",
      "timestamp": "Sep 27, 2020",
    },
    "from": Object {
      "email": "informer@daily.dev",
      "name": "daily.dev",
    },
    "replyTo": Object {
      "email": "hi@daily.dev",
      "name": "daily.dev",
    },
    "templateId": "d-9254665878014627b4fd71593f09d975",
    "to": "lee@acme.com",
    "trackingSettings": Object {
      "openTracking": Object {
        "enable": true,
      },
    },
  },
]
`;
